packages/plugin-ext/src/hosted/node/hosted-plugin-deployer-handler.ts

```diff
@@ -131,6 +131,7 @@ export class HostedPluginDeployerHandler implements PluginDeployerHandler {
             }
 
             const metadata = this.reader.readMetadata(manifest);
+            metadata.isUnderDevelopment = isUnderDevelopment;
```

Comment:
I think that this line suggests that the approach taken here is a little bit against the grain of how the plugin system should work: ideally the plugin reader system should set the metadata, but here `isUnderDelevopment` is grafted onto the metadata after the readers have had their say. Given that, I think the correct approach would be to ensure that whichever `HostedPluginReader` is responsible for setting the metadata of the plugin under development correctly sets this field, rather than passing it through the deployer system.

Reply:
Thank you for the review and the hints to clean up the code. 
Unfortunately the scanner never has access to the `PluginDeployerEntry`.
I now set a marker inside the `PluginDeployerEntryImpl` based on the `originId` and instead of passing the `isUnderDelevopment` through the deployment system, the value is read from the `PluginDeployerEntry`.
What do you think, is this a viable approach?

Reply:
Another alternative - a bit more attractive, though along the same lines as the separate service approach - would be to implement a `MetadataProcessor` for the development host case. Then, at this stage:

https://github.com/eclipse-theia/theia/blob/9b8bc9eb61573ad7e663587af2ef2043371b3321/packages/plugin-ext/src/hosted/node/plugin-reader.ts#L109-L117

The development host metadata processor could add the notation that the plugin in question is under development.